cli/dcoscli/cluster/main.py

```diff
@@ -147,10 +148,22 @@ def _list(json_, attached, linked=False):
     if attached:
         clusters = [c.dict() for c in cluster.get_clusters()
                     if c.is_attached()]
-    elif linked:
-        clusters = [c.dict() for c in cluster.get_linked_clusters()]
     else:
-        clusters = [c.dict() for c in cluster.get_clusters(True)]
+        if linked:
+            clusters = cluster.get_linked_clusters()
+        else:
+            clusters = cluster.get_clusters(True)
+
+        nb_clusters = len(clusters)
+
+        if nb_clusters > 1:
+            # If there are multiple clusters,
+            # query for their versions concurrently.
+            with futures.ThreadPoolExecutor(max_workers=nb_clusters) as pool:
+                reqs = [pool.submit(c.dict) for c in clusters]
+                clusters = [r.result() for r in futures.as_completed(reqs)]
```

Comment:
If we're using `as_completed`, does that mean that the ordering of the results can change every time we fetch them?  If so, I think we probably want to maintain the order, somehow.

Reply:
The order in the table will always be the same (https://github.com/dcos/dcos-cli/blob/beb37ffbbfaf7118ed0e36e0a14a9bec39edf034/cli/dcoscli/tables.py#L920), as for the JSON it might vary. Why do you think that's an issue?

Reply:
Ah ok. I didn't understand that it would be sorted after the fetching was complete.

Reply:
Not when passing the `--json` flag though, let me know if that's a concern and why.

Reply:
it's not a concern